airflow/providers/google/cloud/operators/compute.py

```diff
@@ -240,79 +729,358 @@ def __init__(
+    :param impersonation_chain: Service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :param retry: A retry object used  to retry requests. If `None` is specified, requests
+        will not be retried.
+    :param timeout: The amount of time, in seconds, to wait for the request to complete.
+        Note that if `retry` is specified, the timeout applies to each individual attempt.
+    :param metadata: Additional metadata that is provided to the method.
+    """
+
+    operator_extra_links = (ComputeInstanceTemplateDetailsLink(),)
+
+    # [START gce_instance_template_insert_fields]
+    template_fields: Sequence[str] = (
+        'body',
+        'project_id',
+        'request_id',
+        'gcp_conn_id',
+        'api_version',
+        'impersonation_chain',
+    )
+    # [END gce_instance_template_insert_fields]
+
+    def __init__(
+        self,
+        *,
+        body: dict,
+        project_id: Optional[str] = None,
+        resource_id: Optional[str] = None,
+        request_id: Optional[str] = None,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = (),
+        gcp_conn_id: str = 'google_cloud_default',
+        api_version: str = 'v1',
+        validate_body: bool = True,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ) -> None:
+        self.body = body
+        self.request_id = request_id
+        self.resource_id = self.body["name"] if 'name' in body else resource_id
+        self._field_validator = None  # Optional[GcpBodyFieldValidator]
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+
+        if validate_body:
+            self._field_validator = GcpBodyFieldValidator(
+                GCE_INSTANCE_TEMPLATE_VALIDATION_PATCH_SPECIFICATION, api_version=api_version
+            )
+        self._field_sanitizer = GcpBodyFieldSanitizer(GCE_INSTANCE_FIELDS_TO_SANITIZE)
+        super().__init__(
+            project_id=project_id,
+            zone='global',
+            resource_id=self.resource_id,
+            gcp_conn_id=gcp_conn_id,
+            api_version=api_version,
+            impersonation_chain=impersonation_chain,
+            **kwargs,
+        )
+
+    def check_body_fields(self) -> None:
+        if 'machine_type' not in self.body["properties"]:
```

Comment:
DRY, and we perform the exact same check in the other class as well.

Reply:
Done